nixos/lib/make-iso9660-image.sh

```diff
@@ -29,39 +31,113 @@ addPath() {
 
 stripSlash "$bootImage"; bootImage="$res"
 
-
-if test -n "$bootable"; then
-
-    # The -boot-info-table option modifies the $bootImage file, so
-    # find it in `contents' and make a copy of it (since the original
-    # is read-only in the Nix store...).
-    for ((i = 0; i < ${#targets_[@]}; i++)); do
-        stripSlash "${targets_[$i]}"
-        if test "$res" = "$bootImage"; then
-            echo "copying the boot image ${sources_[$i]}"
-            cp "${sources_[$i]}" boot.img
-            chmod u+w boot.img
-            sources_[$i]=boot.img
-        fi
-    done
-
-    isoBootFlags="-eltorito-boot ${bootImage}
-                  -eltorito-catalog .boot.cat
-                  -no-emul-boot -boot-load-size 4 -boot-info-table
-                  --sort-weight 1 /isolinux" # Make sure isolinux is near the beginning of the ISO
-fi
-
-if test -n "$usbBootable"; then
-    usbBootFlags="-isohybrid-mbr ${isohybridMbrImage}"
+TMP=$(mktemp -d)
+
+mkdir -p append/boot/grub
+mkdir append/EFI
+touch append/boot/grub/device.map
+addPath "/" "$PWD/append"
+
+if test -n "$mbrBootable"; then
+  "$grubMbr/bin/grub-mkstandalone" \
+      --format=i386-pc \
+      --output="$TMP/core.img" \
+      --install-modules="linux normal iso9660 biosdisk memdisk search tar ls configfile" \
+      --modules="linux normal iso9660 biosdisk search" \
+      --locales="" \
+      --fonts="" \
+      "boot/grub/grub.cfg=$grubCfg"
+
+  # TODO: fix this hack to REALLY skip fsprobe
+  "$grubMbr/bin/grub-install" \
+      --target=i386-pc \
+      --boot-directory=$PWD/append/boot \
+      --no-bootsector \
+      --skip-fs-probe \
```

Comment:
this doesn't do it's job properly. would need to fix this, possibly upstream patch